src/core/api/clock.ts

```diff
@@ -345,15 +359,29 @@ export interface IClockOptions {
 function createClock(
   mediaElement : HTMLMediaElement,
   options : IClockOptions
-) : Observable<IClockTick> {
-  return observableDefer(() : Observable<IClockTick> => {
+) : IClockHandler {
+  // Allow us to identify seek performed internally by the player.
+  let isInternalSeekingComing = false;
+  function setCurrentTime(time: number) {
+    isInternalSeekingComing = true;
+    mediaElement.currentTime = time;
+  }
+  const clock$ = observableDefer(() : Observable<IClockTick> => {
     let lastTimings : IClockTick = objectAssign(
       getMediaInfos(mediaElement, "init"),
-      { stalled: null, getCurrentTime: () => mediaElement.currentTime });
-
+      { stalled: null,
+        getCurrentTime: () => mediaElement.currentTime,
+        isInternalSeekingComing: false }
```

Comment:
Why integrating it to a clock tick?
The outside code does not appear to need it